tests/fault_tolerance/cancellation/test_vllm.py

```
@@ -193,10 +170,15 @@ def get_pid(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         """Release allocated port when worker exits."""
         try:
-            # system_port is always allocated in __init__
             deallocate_port(self.system_port)
+            if hasattr(self, "fpm_port"):
+                deallocate_port(self.fpm_port)
+            if hasattr(self, "kv_event_port"):
+                deallocate_port(self.kv_event_port)
+            if hasattr(self, "nixl_side_channel_port"):
+                deallocate_port(self.nixl_side_channel_port)
         except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)